Shared/Extensions/JellyfinAPI/DeviceType.swift

```diff
@@ -0,0 +1,176 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import SwiftUI
+
+enum DeviceType: String, Displayable, SystemImageable, Codable, CaseIterable {
```

Comment:
I'm thinking this may also be a good addition to the SDK 🤔. However since this is now getting large enough I also think we should make this a `struct` with a private init with static instances.

Reply:
I've also only meant to have `SystemImageable` to be used for `SFSymbol` providers. ie: when used in `Image(systemName:)` or `Label(_:systemName:)`.

Instead, make the property return an `ImageResource`.

```swift
var image: ImageResource {
    switch self {
    case .android:
        .deviceAndroid
    // ...
    }
}
```

Reply:
I'm sorry if I'm not understanding this correctly. I have something that works for ImageResource but is this what you meant?
```
    var image: ImageResource {
        ImageResource(
            name: self.rawValue,
            bundle: Bundle.main
        )
    }
```

Reply:
I'd like to use the generated asset enum instead so that we don't need strings for the assets as the raw value of this type.

- https://sarunw.com/posts/swift-symbols-for-asset-catalog/

Reply:
I *think* I did this right. Part of what makes me think I didn't is I couldn't find an extension on Image Resources anywhere? I thought that would be used for Jellyfin-Blue-Blob since I see that used in .jellyfinBlueBlob. I created my own extension and there are no conflicts / duplicates. Let me know if I'm off there!